### PR TITLE
135166 replace token provider with nuget logic

### PR DESCRIPTION
**Pull request by jaceks2106**
Used `LogicTokenProvider` and removed leftover unnecessary token providers

### CODE REVIEW

File: src/ExternalApiExamples/Program.cs [hunk 1]
```diff
@@ -1,15 +1,43 @@
...
+
+        private static async Task Run(AppConfiguration configuration)
         {
             Console.WriteLine("KMD Studica Examples");
-            var tokenProvider = new StudicaHostToHostTokenProvider("-- SECRET --");
+
+            using var httpClient = new HttpClient { BaseAddress = configuration.StudicaExternalApiBaseAddress };
+            httpClient.DefaultRequestHeaders.Add("logic-api-key", configuration.StudicaExternalApiKey);
+
+            var tokenProvider = new LogicTokenProviderFactory(configuration.TokenProvider).GetProvider(httpClient);
```
Comment (jaceks2106) [inline src/ExternalApiExamples/Program.cs:37]: Check if authorizationissuer is assigned

**Review by jaceks2106 — COMMENTED** [review]
It is

File: src/ExternalApiExamples/Program.cs [hunk 2]
```diff
@@ -1,23 +1,52 @@
...
-            await new StudentsExample().Execute(tokenProvider);
-            await new EducationalProgrammesExample().Execute(tokenProvider);
-            await new SubjectCoursesExample().Execute(tokenProvider);
+
+            using var httpClient = new HttpClient();
+            var tokenProvider = new LogicTokenProviderFactory(configuration.TokenProvider).GetProvider(httpClient);
+
+            using var externalApiHttpClient = new HttpClient();
+            externalApiHttpClient.BaseAddress = configuration.StudicaExternalApiBaseAddress;
+            externalApiHttpClient.DefaultRequestHeaders.Add("logic-api-key", configuration.StudicaExternalApiKey);
```
Comment (base2browns) [inline src/ExternalApiExamples/Program.cs]: Shouldn't matter but can we set it to "Logic-Api-Key" please?

File: src/ExternalApiExamples/appsettings.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "TokenProvider": {
+    "AuthorizationScope": "",
+    "ClientId": "",
+    "ClientSecret": ""
+  },
+  "StudicaExternalApiBaseAddress": "",
```
Comment (base2browns) [inline src/ExternalApiExamples/appsettings.json]: We can probably set this to "https://gateway.kmdlogic.io/studica-review/v1"?

File: src/ExternalApiExamples/appsettings.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "TokenProvider": {
+    "AuthorizationScope": "",
```
Comment (base2browns) [inline src/ExternalApiExamples/appsettings.json:3]: We could probably set this. Only problem is it will be different for Prod and Review.

Comment (jaceks2106) [reply src/ExternalApiExamples/appsettings.json:3]: I would leave it up for the tester. Let him do the most of the configuration on his own.

File: src/ExternalApiExamples/Program.cs (hunk 2 above)
Comment (jaceks2106) [reply src/ExternalApiExamples/Program.cs]: fixed

File: src/ExternalApiExamples/appsettings.json (hunk 3 above)
Comment (jaceks2106) [reply src/ExternalApiExamples/appsettings.json]: fixed

**Pull request merged**